modal/_container_io_manager.py

```diff
@@ -393,16 +393,15 @@ async def run_inputs_outputs(self, input_concurrency: int = 1) -> AsyncIterator[
         self._input_concurrency = input_concurrency
         self._semaphore = asyncio.Semaphore(input_concurrency)
 
-        try:
-            async for input_id, function_call_id, input_pb in self._generate_inputs():
-                args, kwargs = self.deserialize(input_pb.args) if input_pb.args else ((), {})
-                self.current_input_id, self.current_input_started_at = (input_id, time.time())
-                yield input_id, function_call_id, args, kwargs
-                self.current_input_id, self.current_input_started_at = (None, None)
-        finally:
-            # collect all active input slots, meaning all inputs have wrapped up.
-            for _ in range(input_concurrency):
-                await self._semaphore.acquire()
+        async for input_id, function_call_id, input_pb in self._generate_inputs():
+            args, kwargs = self.deserialize(input_pb.args) if input_pb.args else ((), {})
+            self.current_input_id, self.current_input_started_at = (input_id, time.time())
+            yield input_id, function_call_id, args, kwargs
+            self.current_input_id, self.current_input_started_at = (None, None)
+
+        # collect all active input slots, meaning all inputs have wrapped up.
+        for _ in range(input_concurrency):
```

Comment:
Fix 1: Just remove the finally clause and let exceptions from `_generate_inputs` just bubble up. I don't think this should have any crazy effects since we probably want to exit here anyway and it shouldn't matter/should be intentional if some ongoing inputs are dropped?